src/main/kotlin/algorithms/FindBridgesAlgorithm.kt

```diff
@@ -0,0 +1,82 @@
+package algorithms
+
+import graph.UndirectedAdjacencyList
+import java.util.Stack
+import kotlin.math.min
+
+class FindBridgesAlgorithm(private val adjacencyList: UndirectedAdjacencyList) {
+    private val verticesCount = adjacencyList.verticesCount()
+    private val verticesStack = Stack<Int>()
```

Comment:
Why is it 'verticesStack' ? 

You took it from my algorithm I think. But that stack was for algorithm specified purposes. Stack for  nonrecursive dfs implementation is named 'dfsStack'

Reply:
fixed!